src/modules/roc_core/target_libunwind/roc_core/backtrace.cpp

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#define UNW_LOCAL_ONLY
+#include <libunwind.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "roc_core/backtrace.h"
+
+namespace roc {
+namespace core {
+
+namespace {
+
+enum { MaxDigits = 10, MaxLen = 200, MaxLenFunctionName = 128 };
+/* safe concatenation of strings
+ */
+void safe_strcat(char* buffer, size_t& buffer_size, const char* str) {
+    size_t len = strlen(str);
+    /* Checking if there is enough space in the buffer
+     */
+    if (len < MaxLen - buffer_size - 1) {
+        len = MaxLen - buffer_size - 1;
+    }
+    if (len > 0) {
+        strncat(buffer, str, len);
```

Comment:
`strncat(buffer, str, len)` is incorrect. It should be `strncat(buffer + buffer_size, str, len)`.

Also, GCC 8 produces a warning on this piece:

```
error: 'char* strncat(char*, const char*, size_t)'
specified bound 3 equals source length [-Werror=stringop-overflow=]
         strncat(buffer + buffer_size, str, len);
```

Here is how we can suppress the warning:

```
        memcpy(buffer + buffer_size, str, len);
        buffer_size += len;
        buffer[buffer_size] = '\0';
```